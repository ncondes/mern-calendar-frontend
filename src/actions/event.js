import { fetchWithToken } from '../helpers/fetch';
import { types } from '../types/types';
import { prepareEvents } from '../helpers/prepareEvents';
import Swal from 'sweetalert2';



export const eventStartAddNew = ( event ) => {
    return async ( dispatch, getState ) => {

        const { uid, name } = getState().auth;

        try {

            const res = await fetchWithToken( 'events', event, 'POST' );
            const body = await res.json();
    
            if ( body.ok ) {

                event.id = body.event.id;
                event.user = {
                    _id: uid,
                    name,
                }

                console.log( event );
                dispatch( eventAddNew( event ) );
            }

        } catch ( error ) {
            console.log( error );
        }


    }
}


const eventAddNew = ( event ) => ({
    type: types.eventAddNew,
    payload: event,
});

export const eventSetActive = ( event ) => ({
    type: types.eventSetActive,
    payload: event,
});

export const eventClearActiveEvent = () => ({
    type: types.eventClearActiveEvent,
});


export const eventStartDelete = () => {
    return async ( dispatch, getState ) => {

        const { id } = getState().calendar.activeEvent;

        try {
            
            const res = await fetchWithToken( `events/${ id }`, {}, 'DELETE' );
            const body = await res.json();

            if ( body.ok ) {
                dispatch( eventDeleted() );
            } else {
                Swal.fire('Error', body.msg, 'error');
            }

        } catch ( error ) {
            console.log( error );
        }

    }
}


const eventDeleted = () => ({
    type: types.eventDeleted,
});


export const eventStartLoading = () => {
    return async ( dispatch ) => {

        try {
            
            const res = await fetchWithToken( 'events' );
            const body = await res.json();
            const events = prepareEvents( body.events );

            dispatch( eventLoaded( events ) )

        } catch ( error ) {
            console.log( error );
        }

    }
}

const eventLoaded = ( events ) => ({
    type: types.eventLoaded,
    payload: events,
})



export const eventStartUpdate = ( event ) => {
    return async ( dispatch ) => {

        try {
            
            const res = await fetchWithToken( `events/${ event.id }`, event, 'PUT' );
            const body = await res.json();

            if ( body.ok ) {
                dispatch( eventUpdated( event ) );
            } else {
                Swal.fire('Error', body.msg, 'error');
            }

        } catch ( error ) {
            console.log( error );
        }

    }
}


const eventUpdated = ( event ) => ({
    type: types.eventUpdated,
    payload: event,
});


export const eventLogout = () => ({
    type: types.eventLogout,
})