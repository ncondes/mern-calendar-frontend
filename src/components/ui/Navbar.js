

export const Navbar = () => {
    return (
        <nav className="navbar navbar-dark bg-dark mb-2">
            <div className="container-fluid">
                <span className="navbar-brand">Roberto</span>
                <button className ="btn btn-outline-light" type ="submit">
                    <i className='fas fa-sign-out-alt'></i>
                    <span className=''> Log out</span>
                </button>
            </div>
        </nav>
    )
}
