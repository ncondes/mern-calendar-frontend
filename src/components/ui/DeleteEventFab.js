import { useDispatch } from 'react-redux';
import { eventDeleted } from '../../actions/event';


export const DeleteEventFab = () => {

    const dispatch = useDispatch();

    const handleDelete = () => {
        dispatch( eventDeleted() );
    }

    return (
        <button
            className='btn btn-danger fab-danger'
            onClick= { handleDelete }
        >
            <i className='fas fa-trash'></i>
            <span> Delete Event</span>
        </button>
    )
}
