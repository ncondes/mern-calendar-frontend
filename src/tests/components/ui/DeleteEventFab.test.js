
import { mount } from 'enzyme'
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import '@testing-library/jest-dom';
import { DeleteEventFab } from '../../../components/ui/DeleteEventFab';
import { eventStartDelete } from '../../../actions/event';



// mock a la funcion
jest.mock('../../../actions/event', () => ({
    eventStartDelete: jest.fn(),
}))


// config store
const middlewares = [ thunk];
const mockStore = configureStore( middlewares );

const initState = {};
let store = mockStore( initState );

// mock store
store.dispatch = jest.fn();

const wrapper = mount(
    <Provider store={ store }>
        <DeleteEventFab />
    </Provider>
)



describe('Pruebas en DeleteEventFab', () => {

    test('Debe de hacer match con el snapshot', () => {
        
        expect( wrapper ).toMatchSnapshot();

    });

    test('Debe de llamar el eventStartDelete al hacer click', () => {

        wrapper.find('.fab-danger').prop('onClick')();
        expect( eventStartDelete ).toHaveBeenCalled();

    })
    
    
}) 