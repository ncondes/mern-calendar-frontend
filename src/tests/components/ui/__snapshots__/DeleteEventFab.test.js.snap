// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en DeleteEventFab Debe de hacer match con el snapshot 1`] = `
<button
  className="btn btn-danger fab-danger"
  onClick={[Function]}
>
  <i
    className="fas fa-trash"
  />
  <span>
     Delete Event
  </span>
</button>
`;
